packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageView.kt

```diff
@@ -0,0 +1,188 @@
+      eventsManager.onLoadStarted()
+      requestManager
+        .load(sourceToLoad)
+        .apply(options)
+        .listener(eventsManager)
+        .into(this)
+      requestManager
+        .`as`(BitmapFactory.Options::class.java)
+        .load(sourceToLoad)
+        .into(eventsManager)
+    }
+  }
+
+  internal fun onDrop() {
+    requestManager.clear(this)
+  }
+
+  private fun createUrlFromSourceMap(sourceMap: ReadableMap?): GlideUrl? {
+    val uriKey = sourceMap?.getString(SOURCE_URI_KEY)
+    return uriKey?.let { GlideUrl(uriKey) }
+  }
+
+  private fun createOptionsFromSourceMap(sourceMap: ReadableMap?): RequestOptions {
+    return RequestOptions()
+      .fitCenter()
+      .apply {
+        // Override the size for local assets. This ensures that
+        // resizeMode "center" displays the image in the correct size.
+        if (sourceMap != null
+          && sourceMap.hasKey(SOURCE_WIDTH_KEY)
+          && sourceMap.hasKey(SOURCE_HEIGHT_KEY)
+          && sourceMap.hasKey(SOURCE_SCALE_KEY)) {
+
+          val scale = sourceMap.getDouble(SOURCE_SCALE_KEY)
+          val width = sourceMap.getInt(SOURCE_WIDTH_KEY)
+          val height = sourceMap.getInt(SOURCE_HEIGHT_KEY)
+          override((width * scale).toInt(), (height * scale).toInt())
+        }
+      }
+  }
+
+  private fun getOrCreateBorderDrawable(): BorderDrawable {
+    if (borderDrawable == null) {
+      borderDrawable = BorderDrawable(context)
+      borderDrawable!!.callback = this
+      val borderRadii = outlineProvider.borderRadiiConfig
+      for (i in borderRadii.indices) {
+        var borderRadius = borderRadii[i]
+        borderRadius = if (!YogaConstants.isUndefined(borderRadius)) PixelUtil.toPixelFromDIP(borderRadius) else borderRadius
+        if (i == 0) {
+          borderDrawable!!.setRadius(borderRadius)
+        } else {
+          borderDrawable!!.setRadius(borderRadius, i - 1)
+        }
+      }
+    }
+    return borderDrawable!!
+  }
+
+  // Drawing overrides
+  override fun invalidateDrawable(drawable: Drawable) {
+    super.invalidateDrawable(drawable)
+    if (drawable === borderDrawable) {
+      invalidate()
+    }
+  }
+
+  override fun draw(canvas: Canvas) {
+
+    // When the border-radii are not all the same, a convex-path
+    // is used for the Outline. Unfortunately clipping is not supported
+    // for convex-paths and we fallback to Canvas clipping.
+    outlineProvider.clipCanvasIfNeeded(canvas, this)
+    super.draw(canvas)
+  }
+
+  public override fun onDraw(canvas: Canvas) {
+    super.onDraw(canvas)
+
+    // Draw borders on top of the background and image
+    if (borderDrawable != null) {
+      val layoutDirection = if (I18nUtil.getInstance().isRTL(context)) LAYOUT_DIRECTION_RTL else LAYOUT_DIRECTION_LTR
+      borderDrawable!!.setResolvedLayoutDirection(layoutDirection)
+      borderDrawable!!.setBounds(0, 0, canvas.width, canvas.height)
```

Comment:
Android Studio marks `canvas.width, canvas.height` as misused here:
> Referencing Canvas.width is usually wrong; you should be referencing width instead

Is it correct?

Reply:
Yeah, let's trust in IntelliJ 😄 
```suggestion
      borderDrawable!!.setBounds(0, 0, width, height)
```